### ISSUE TITLE
Feature: `/nmui-config.js` could be served directly from Caddyfile

### ISSUE THREAD

**Issue by nazarewk** [gravitl/netmaker#2775]
While working on a NixOS module I noticed Caddy is able to generate responses based on environment variables by itself using snippet like this:

https://github.com/nazarewk-iac/nix-configs/blob/a92e06d3a2d04e7c034c9f1f54f23dc76a387580/packages/netmaker/netmaker-ui/Caddyfile

If you replaced `nginx` with `caddy`, you could get rid of https://github.com/gravitl/netmaker-ui-2/blob/b305a96fc19160346f7cd59439d4625f262aaedf/generate-config.sh

**Comment by Aceix** [comment]
interesting @nazarewk. when does this script run to set the vars? https://github.com/nazarewk-iac/nix-configs/blob/a92e06d3a2d04e7c034c9f1f54f23dc76a387580/packages/netmaker/netmaker-ui/netmaker-ui.sh

**Comment by nazarewk** [comment]
> interesting @nazarewk. when does this script run to set the vars? [nazarewk-iac/nix-configs@`a92e06d`/packages/netmaker/netmaker-ui/netmaker-ui.sh](https://github.com/nazarewk-iac/nix-configs/blob/a92e06d3a2d04e7c034c9f1f54f23dc76a387580/packages/netmaker/netmaker-ui/netmaker-ui.sh)

this one is installed as a `netmaker-ui` binary starting up the whole thing:
https://github.com/nazarewk-iac/nix-configs/blob/a92e06d3a2d04e7c034c9f1f54f23dc76a387580/packages/netmaker/netmaker-ui/default.nix#L20

**Comment by Aceix** [comment]
@nazarewk got it. but your setup can still work with how we currently use nginx right?

**Comment by nazarewk** [comment]
I'm not using netmaker anymore for a few months after it stopped being usable in free version, but I remember it was generally easier to work with Caddy's interpolation due to read-only filesystems throughout NixOS

**Comment by Aceix** [comment]
right. then we may consider switching to Caddy later. will close this ticket for now
